metagraph/src/annotation/binary_matrix/base/binary_matrix.cpp

```diff
@@ -32,6 +34,17 @@ BinaryMatrix::slice_rows(const std::vector<Row> &row_ids) const {
     return slice;
 }
 
+void BinaryMatrix::slice_columns(const std::vector<Column> &column_ids,
+                                 const std::function<void(Column, bitmap&&)> &callback,
+                                 size_t /* num_threads */) const {
+    size_t nrows = num_rows();
+
+    for (size_t k = 0; k < column_ids.size(); ++k) {
+        Column j = column_ids[k];
+        callback(j, bitmap_generator(get_column(j), nrows));
+    }
```

Comment:
why not enable `num_threads`?
```suggestion
                                 size_t num_threads) const {
    #pragma omp parallel for num_threads(num_threads) schedule(dynamic)
    for (size_t k = 0; k < column_ids.size(); ++k) {
        Column j = column_ids[k];
        callback(j, bitmap_generator(get_column(j), num_rows()));
    }
```

Reply:
The problem here is that `get_column` can return a huge vector, so if a large number of threads is passed, this can easily blow up the RAM usage.

Reply:
You can always set num_threads to 1 in the caller if memory is an issue.